.circleci/config.yml

```diff
@@ -174,24 +177,65 @@ commands:
       py-version:
         <<: *py-version-template
     steps:
+      - run:
+          name: Preparing environment
+          command: |
+            export OS_NAME=$(uname -s | sed s/Darwin/macos/ | tr '[:lower:]' '[:upper:]')
+            echo "export OS_NAME=$OS_NAME" >> ${BASH_ENV}
+            if [[ ${OS_NAME} == "MACOS" ]]; then
+              echo "export PYENV_ROOT=~/.local-MACOS/.pyenv" >> ${BASH_ENV}
+              if [[ $PYTHON_VERSION_SHORT == 3.8 ]]; then
+                export PYENV_PYTHON_VERSION=3.8.6
+              elif [[ $PYTHON_VERSION_SHORT == 3.7 ]]; then
+                export PYENV_PYTHON_VERSION=3.7.9
+              else
+                exit 1
+              fi
+              echo "export PYENV_PYTHON_VERSION=$PYENV_PYTHON_VERSION" >> ${BASH_ENV}
+            fi
+            echo "SHA: ${CIRCLE_SHA1}"
+            echo "TAG: ${CIRCLE_TAG}"
       - attach-and-link:
           py-version: << parameters.py-version >>
-      - config-path
+      - restore_cache:
+          keys:
+          - system-deps-pyenv-v4-{{ arch }}-<< parameters.py-version >>
+      - run:
+          name: Installing Python on MacOS
+          command: |
+            if [[ ${OS_NAME} == "MACOS" ]]; then
+              if [[ ! -x $PYENV_ROOT ]]; then
+                git clone https://github.com/pyenv/pyenv.git $PYENV_ROOT
+              fi
+              export PATH=${PYENV_ROOT}/bin:${PATH}
+              env PYTHON_CONFIGURE_OPTS="--enable-shared" pyenv install $PYENV_PYTHON_VERSION -sv
```

Comment:
Does pyenv automatically skip building if an existing version is found?

Reply:
No, but with the `-s` flag it does ;)

Reply:
Nice!